02_bubble/alextmz/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+
+	for len(s) > 0 {
+		for k, v := range s {
```

Comment:
This is not bubble sort as I know it. Without early bail it should look something like this:
```go
	for i := len(s); i > 0; i-- {
		for j := 1; j < i; j++ {
			if s[j-1] > s[j] {
				s[j-1], s[j] = s[j], s[j-1]
			}
		}
        }
```
What does len(s) > 0 mean as the loop condition with s not being modified? Shouldn't it just be a forever? `for { ...`

Also, you are meant to sort a copy and not modify the input (see [lab 2 assignment](https://github.com/anz-bank/go-course/blob/master/02_bubble/README.md)).

Reply:
Yep, I've tried to do a bubblesort from head and kinda focused more on the tests, so it was awfully wrong. Tests were very wrong too btw! :P All fixed now. Thanks for the review!